go-runtime/ftl/ftltest/fake.go

```diff
@@ -121,3 +181,236 @@ func actuallyCallMap(ctx context.Context, impl mapImpl) any {
 	}
 	return out
 }
+
+func (f *fakeFTL) PublishEvent(ctx context.Context, topic *schema.Ref, event any) error {
+	f.publishWaitGroup.Add(1)
+	f.globalTopic.PublishSync(publishEvent{topic: topic, content: event})
+	return nil
+}
+
+// addSubscriber adds a subscriber to the fake FTL instance. Each subscriber included in the test must be manually added
+func addSubscriber[E any](f *fakeFTL, sub ftl.SubscriptionHandle[E], sink ftl.Sink[E]) {
+	f.pubSubLock.Lock()
+	defer f.pubSubLock.Unlock()
+
+	if _, ok := f.subscriptions[sub.Name]; !ok {
+		f.subscriptions[sub.Name] = &subscription{
+			name:   sub.Name,
+			topic:  sub.Topic,
+			errors: map[int]error{},
+		}
+	}
+
+	subscribers, ok := f.subscribers[sub.Name]
+	if !ok {
+		subscribers = []subscriber{}
+	}
+	f.subscribers[sub.Name] = append(subscribers, func(ctx context.Context, event any) error {
```

Comment:
```suggestion
	f.subscribers[sub.Name] = append(f.subscribers[sub.Name], func(ctx context.Context, event any) error {
```

^^ this is equivalent